### PR TITLE
Feat/key headers

### PR DESCRIPTION
**Pull request by andrewstevenson**
Add KCQL fields for keys and headers

* if field begins with `_key` add to keyFields
* if field begins with `_header` add to headerFields

### CODE REVIEW

File: src/main/java/com/datamountaineer/kcql/Kcql.java [hunk 1]
```diff
@@ -190,6 +250,16 @@ public String getTarget() {
     //return new ArrayList<>(fields);
   }
 
+  public List<Field> getKeyFields() {
+    return keyFields;
+    //return new ArrayList<>(fields);
```
Comment (davidsloan) [inline src/main/java/com/datamountaineer/kcql/Kcql.java:255]: minor, but we should clean up this commented code (also below)

File: src/main/java/com/datamountaineer/kcql/Kcql.java [hunk 2]
```diff
@@ -516,7 +586,25 @@ public void exitColumn_name(ConnectorParser.Column_nameContext ctx) {
         if (isWithinIgnore[0]) {
           kcql.ignoredFields.add(field);
         } else {
-          kcql.addField(field);
+          List<String> cleanedParent = null;
```
Comment (davidsloan) [inline src/main/java/com/datamountaineer/kcql/Kcql.java:589]: I think `cleanedParent` might be unnecessary? Can we not just use `nestedFieldsBuffer` directly?

**Review by davidsloan — APPROVED** [review]
A few minor points, nothing to get excited about.

File: src/main/java/com/datamountaineer/kcql/Kcql.java (hunk 1 above)
Comment (andrewstevenson) [reply src/main/java/com/datamountaineer/kcql/Kcql.java]: cleaned up

File: src/main/java/com/datamountaineer/kcql/Kcql.java (hunk 2 above)
Comment (andrewstevenson) [reply src/main/java/com/datamountaineer/kcql/Kcql.java:585]: updated to set the parent as null if root element

**Pull request merged**